include/core/PyHydrofabricSubsetter.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef NGEN_PYHYDROFABRICSUBSETTER_HPP
+#define NGEN_PYHYDROFABRICSUBSETTER_HPP
+
+#ifdef ACTIVATE_PYTHON
+#include <pybind11/embed.h>
+namespace py = pybind11;
+#endif // ACTIVATE_PYTHON
```

Comment:
Any reason not to wrap this entire header in the `#ifdef ACTIVATE_PYTHON` macro?  The way this class is now, you can create a valid `PyHydroFabricSubsetter` if `ACTIVATE_PYTHON` is False, but that object just has a few path strings in it and doesn't do anything...

Reply:
I think the answer to the first question is "yes." 

In a situation like this, where the type needs a file to function, I think throwing an exception in the constructor is the appropriate way to handle if the file is not available.  We could defer this kind of check until after the constructor I suppose, but I'm not sure that's really any better.

Regardless, sometimes the possibility of not being able to obtain a useable instance of a type is unavoidable, in particular when files are involved, but I expect in other situations also.

Given that, should using code be concerned with determining whether it _will_ get back a usable instance, when either way it is going to need to be concerned with whether it _did_ get back one?

A caveat and counter argument, however:  even assuming that approach is better in general, it is better in this case?  Right now, our using code is already necessarily pretty concerned with both whether Python is active and whether those files are available.

Reply:
I went ahead and went with the simpler approach for now.  There are several things in the driver and driver utilities that need to be improved, so deciding on this probably should wait until then.